tests/robust/test_internals_linearize.py

```diff
@@ -178,3 +174,173 @@ def test_nmc_param_influence_vmap_smoke(
         assert not torch.isnan(v).any(), f"eif for {k} had nans"
         assert not torch.isinf(v).any(), f"eif for {k} had infs"
         assert not torch.isclose(v, torch.zeros_like(v)).all(), f"eif for {k} was zero"
+
+
+@pytest.mark.parametrize(
+    "loc", [torch.zeros(2, requires_grad=True), torch.ones(2, requires_grad=True)]
+)
+@pytest.mark.parametrize(
+    "cov_mat",
+    [
+        torch.eye(2, requires_grad=False),
+        torch.tensor(torch.ones(2, 2) + torch.eye(2), requires_grad=False),
+    ],
+)
+@pytest.mark.parametrize(
+    "v",
+    [
+        torch.tensor([1.0, 0.0], requires_grad=False),
+        torch.tensor([0.0, 1.0], requires_grad=False),
+        torch.tensor([1.0, 1.0], requires_grad=False),
+        torch.tensor([0.0, 0.0], requires_grad=False),
+    ],
+)
+def test_empirical_fisher_vp_against_analytical(
+    loc: torch.Tensor, cov_mat: torch.Tensor, v: torch.Tensor
+):
+    func_log_prob = gaussian_log_prob
+    log_prob_params = {"loc": loc}
+    N_monte_carlo = 10000
+    data = Predictive(GaussianModel(cov_mat), num_samples=N_monte_carlo)(loc)
+    empirical_fisher_vp_func = make_empirical_fisher_vp(
+        func_log_prob, log_prob_params, data, cov_mat=cov_mat
+    )
+
+    empirical_fisher_vp = empirical_fisher_vp_func({"loc": v})["loc"]
+
+    prec_matrix = torch.linalg.inv(cov_mat)
+    true_vp = prec_matrix.mv(v)
+
+    assert torch.all(torch.isclose(empirical_fisher_vp, true_vp, atol=0.1))
+
+
+@pytest.mark.parametrize(
+    "data_config",
+    [
+        (torch.zeros(1, requires_grad=True), torch.eye(1)),
+        (torch.ones(2, requires_grad=True), torch.eye(2)),
+    ],
+)
+def test_fisher_vmap_smoke(data_config):
```

Comment:
I don't think this should just be a smoke test - with `vmap`, you can check that the `vmap`-ed output matches the output of evaluating the function on each item in the batch sequentially.